spec/type_references_and_ownership.md

```diff
@@ -0,0 +1,503 @@
+
+- Embed the dependent resource directly in the same ARM template in which the owning resource is created, using the `resources` property.
+    - This requires that the owning resource and the dependent resource are created at the same time.
+- Deploy the dependent resource separately.
+    - When doing this, you should specify `dependsOn` stating that your new dependent resource depends on its owner. This is not strictly required in cases where
+      the owner already exists, but is required when it is still being created (possibly in parallel).
+    - The `name` field of the dependent resource in this mode serves a dual purpose - to name the dependent resource and to identify the parent resources.
+      Each segment of the name corresponds to an owning resource, so creating a Batch Pool `foo` in Batch account `account` would have `name` = `account/foo`.
+
+#### What ASO does
+Dependent resources in ASO have properties which map to the name/path to their owner. For example 
+[MySQLFirewallRuleSpec](https://github.com/Azure/azure-service-operator/blob/92240406aff3863f3a267d8a1dc1e28aa3e841ae/api/v1alpha1/mysqlfirewallrule_types.go#L15) looks like this:
+
+```go
+type MySQLFirewallRuleSpec struct {
+	ResourceGroup  string `json:"resourceGroup"`
+	Server         string `json:"server"`
+	StartIPAddress string `json:"startIpAddress"`
+	EndIPAddress   string `json:"endIpAddress"`
+}
+```
+
+The `ResourceGroup` and `Server` are references to the owners of this type.
+
+#### What k8s-infra does
+k8s-infra uses the same `KnownTypeReference` type mentioned above for ownership references too.
+There are two patterns for ownership in k8s-infra today.
+
+One pattern is used for ResourceGroup, where 
+[top level resources have a link to the resource group they are in](https://github.com/Azure/k8s-infra/blob/14105b1cb3f6967cd086c9f8f75fb16bb85d6318/apis/microsoft.compute/v1/virtualmachinescaleset_types.go#L323).
+
+```go
+type VirtualMachineScaleSetSpec struct {
+    // +kubebuilder:validation:Required
+    ResourceGroupRef *azcorev1.KnownTypeReference `json:"resourceGroupRef" group:"microsoft.resources.infra.azure.com" kind:"ResourceGroup"`
+    ...
+}
+```
+
+The other pattern is where the owning resource has links to the dependent resources it is expecting to have:
+
+```go
+type RouteTableSpecProperties struct {
+    DisableBGPRoutePropagation bool                          `json:"disableBgpRoutePropagation,omitempty"`
+    RouteRefs                  []azcorev1.KnownTypeReference `json:"routeRefs,omitempty" group:"microsoft.network.infra.azure.com" kind:"Route" owned:"true"`
+}
+```
+
+If the dependent resources aren't there, the status of the owning resource reflects an error explaining that.
+
+## Proposal
+
+### How to represent references
+We construct a new type which is our reference holder. I am calling this approach `AugmentedKnownTypeReference` (real type name TBD), because
```

Comment:
How about `ResourceReference` ?

And should it be an interface, implemented by both `KubernetesReference` and `AzureReference` - or does this not show well in the CRD?